### ISSUE TITLE
[RLlib] PPO nan reward for many rollout workers

### ISSUE THREAD

**Issue by mlpeschl** [ray-project/ray#37514]
### What happened + What you expected to happen

I am running the simple PPO cartpole example from the [PPO examples](https://docs.ray.io/en/latest/rllib/rllib-algorithms.html#ppo). It seems that increasing the number of rollout workers and number of environments per worker can lead to nan values in the episode rewards.

I have tried playing around with the values and have noticed that reducing the learning rate, as well as the number of sgd steps in PPO can counteract for the increase in rollout workers and environments. However, this dramatically slows down learning. However, akin to a scalable implementation of PPO, I would expect the possibility to distribute the load of acting in the environment to many (for example [thousands for Dota](https://arxiv.org/pdf/1912.06680.pdf)) workers, while the GPU handles the forward passes.

### Versions / Dependencies

Ray v2.5.0
Python 3.8.10

### Reproduction script

```python
from ray.rllib.algorithms.ppo import PPOConfig
from ray import air
from ray import tune


config = PPOConfig()
# Print out some default values.
print(config.clip_param)  
# Update the config object.
config.training(lr=0.001)
# Set the config object's env.
config = config.environment(env="CartPole-v1")
config = config.framework("torch") 
config = config.rollouts(num_envs_per_worker=256, num_rollout_workers=6)
config = config.resources(num_gpus=1)
# Use to_dict() to get the old-style python config dict
# when running with tune.
tune.Tuner(  
    "PPO",
    run_config=air.RunConfig(stop={"episode_reward_mean": 200}),
    param_space=config.to_dict(),
).fit()
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by ArturNiederfahrenhorst** [comment]
Can you post the error that you get and paste it here?
You can't increase `num_envs_per_worker` and `num_rollout_workers` "infinitely.
The throughput will also be bound by the GPU and batch size, minibatch size and number of sgd iterations also influence the dynamics between sampling throughput and learning throughput greatly.

**Comment by ArturNiederfahrenhorst** [comment]
Please experiment with these parameters and see what that does.

**Comment by mlpeschl** [comment]
Hi Artur! Thank you for the quick response. 

There is no specific error I am receiving, but the ray tune outputs log NaN values for the episode rewards as follows:

```
Number of trials: 1/1 (1 RUNNING)
+-----------------------------+----------+-----------------------+--------+------------------+------+----------+----------------------+----------------------+--------------------+
| Trial name                  | status   | loc                   |   iter |   total time (s) |   ts |   reward |   episode_reward_max |   episode_reward_min |   episode_len_mean |
|-----------------------------+----------+-----------------------+--------+------------------+------+----------+----------------------+----------------------+--------------------|
| PPO_CartPole-v1_2e714_00000 | RUNNING  | 10.231.137.78:4109470 |      1 |          18.3534 | 7680 |      nan |                  nan |                  nan |                nan |
+-----------------------------+----------+-----------------------+--------+------------------+------+----------+----------------------+----------------------+--------------------+
```

Your suggestion of changing the batch and minibatch size have cleared my confusion with respect to what is happening.  Namely, setting the `batch_mode `to complete episodes stops showing nan values. I suppose nan is shown because there first few iterations, PPO will update using partial trajectories from incomplete episodes only. Maybe changing `nan` to `None` (or emitting a warning in this case) would have been less confusing (but this is of course personal preference).

**Comment by ArturNiederfahrenhorst** [comment]
Glad it's working now. Thanks for the feedback! I'll keep that in the back of my head for now.
